README
			   * nemo-fileroller *

		  File Roller extension for nemo


* Licensing

  This program is released under the terms of the GNU General Public
  License (GNU GPL) version 2 or greater.
  You can find a copy of the license in the file COPYING.

* Dependencies

  In order to build this program from the source code you need a working
  GNOME environment version 2.x, with the development tools installed
  properly.

  Also you need the following libraries:
  	* glib >= 2.16.0
  	* gtk+ >= 2.12.0
  	* libgnome >= 2.6.0
  	* libgnomeui >= 2.6.0
        * libglade >= 2.4.0
        * libnemo-extension >= 2.22.2 (optional)

* Install

  ./configure
  make
  make install


ChangeLog
commit cd12c5b1cf03b9eb47b07f16ab5797e82e86249d
Author: Paolo Bacchilega <paobac@src.gnome.org>
Date:   2012-05-14

    updated for version 3.4.2

M	NEWS
M	configure.ac

commit cad919c516fe08355bd6ba9a6cfffb126fdd3e4c
Author: Tom Tryfonidis <tomtryf@gmail.com>
Date:	2012-05-02

    Updated Greek translation

M	po/el.po

commit 511e0b3723c990939806b3b6df0d6f4ad79e698e
Author: Takayuki KUSANO <AE5T-KSN@asahi-net.or.jp>
Date:	2012-04-24

    Updated Japanese translation.

M	po/ja.po

commit 88a58e214579e18610fa9227736fc75dfdb7b62b
Author: Paolo Bacchilega <paobac@src.gnome.org>
Date:	2012-04-16

    updated for version 3.4.1

M	NEWS
M	configure.ac

commit 3d298bb1ebc0ff6c6a0b0c30e714dbf7313ba0df
Author: Mark Kwidzińsczi <Unknown>
Date:	2012-04-16

    Added Kashubian translation

M	po/LINGUAS
A	po/csb.po

commit 9741771855d9d90d126e4dc403e145d9e35ffd39
Author: Peter Mráz <etkinator@gmail.com>
Date:	2012-04-16

    Updated Slovak translation

M	po/sk.po

commit ebb56bc7a5e001963e54d7d90a1abe622320ca0e
Author: Daniel Martinez Cucalon <dmartinez@src.gnome.org>
Date:	2012-04-13

    Updated Aragonese translation

M	po/an.po

commit 78c8bb6d5a3860c5fba64490e55aff633b4eb6d1
Author: Carles Ferrando <carles.ferrando@gmail.com>
Date:	2012-04-10

    [l10n]Updated Catalan (Valencian) translation

M	po/ca@valencia.po

commit 6899e4845c8a0b0145e827a53d98cbbaf60587e6
Author: Jordi Serratosa <jordis@softcatala.cat>
Date:	2012-04-10

    [l10n] Fixes on Catalan translation

M	po/ca.po

commit 1da8c34f1209e6be07811e7ef0cd19a4ca08afcb
Author: Andrej Žnidaršič <andrej.znidarsic@gmail.com>
Date:	2012-04-09

    Updated Slovenian translation

M	help/sl/sl.po

commit dad7bc5f5f957c7b858df0c6598bc80690973dd4
Author: Michael Kotsarinis <mk73628@gmail.com>
Date:	2012-04-01

    Updated Greek translation

M	po/el.po

commit b0b5cb22a5d6ef144a11630022ba149ca506a0a4
Author: Andika Triwidada <andika@gmail.com>
Date:	2012-04-01

    [l10n] Updated Indonesian translation

M	po/id.po

commit b21d15d025735a5a7f844358624d48c2f868c356
Author: Daniel Mustieles <daniel.mustieles@gmail.com>
Date:	2012-03-27

    Updated Spanish translation

M	po/es.po

commit 0fe6b5abd15291b8fab9f13ad3f0398299b21548
Author: Daniel Mustieles <daniel.mustieles@gmail.com>
Date:	2012-03-27

    Updated Spanish translation

M	help/es/es.po

commit 1a6dc3c3d6d159765c50eb2b58b4a798be55d3fd
Author: Daniel Mustieles <daniel.mustieles@gmail.com>
Date:	2012-03-27

    Updated spanish translation

M	help/es/es.po

commit 5ca51cd7f54db5132b1123961f64d896959e431d
Author: Daniel Mustieles <daniel.mustieles@gmail.com>
Date:	2012-03-27

    Updated spanish translation

M	help/es/es.po

commit cd396677068ec078e796fd1cf07c2015b764cb0e
Author: Daniel Mustieles <daniel.mustieles@gmail.com>
Date:	2012-03-27

    Updated spanish translation

M	po/es.po

commit 0fd4c3eb5959caf138d1653ed9a33486f46ecb91
Author: Daniel Mustieles <daniel.mustieles@gmail.com>
Date:	2012-03-27

    Updated spanish translation

M	po/es.po

commit dd59ef3fae5267183e102c467a3e31e4e2e7ba79
Author: Paolo Bacchilega <paobac@src.gnome.org>
Date:	2012-03-26

    updated for version 3.4.0

M	NEWS
M	configure.ac

commit cea10d17b146e30585bdbf8bbd06e8b0af821419
Author: Khaled Hosny <khaledhosny@eglug.org>
Date:	2012-03-26

    Updated Arabic translation

M	po/ar.po

commit f9cde1299d01b2090c2fb6e8a3c0e07c9e8ba9a3
Author: Bruno Brouard <annoa.b@gmail.com>
Date:	2012-03-26

    Updated French doc translation

M	help/fr/fr.po

commit 271f5b8805131c147a4690303654334a96b3198a
Author: Yasumichi Akahoshi <yasumichi@vinelinux.org>
Date:	2012-03-26

    [l10n] Update Japanese translation

A	help/ja/figures/file-roller-main-window.png
D	help/ja/figures/file-roller_home.png
D	help/ja/figures/file-roller_leftarrow.png
D	help/ja/figures/file-roller_main_window.png
D	help/ja/figures/file-roller_rightarrow.png
D	help/ja/figures/file-roller_uparrow.png
M	help/ja/ja.po

commit 794dfe9faba7fe25e93fca2d6886e363c05a53d8
Author: Gil Forcada <gforcada@gnome.org>
Date:	2012-03-26

    [l10n] Minor cosmetic change on Catalan translation

M	po/ca.po

commit 1a7e6260eb817d05c6f078b5c9509794105898bd
Author: Carles Ferrando <carles.ferrando@gmail.com>
Date:	2012-03-26

    [l10n]Updated Catalan (Valencian) translation

M	po/ca@valencia.po

commit a379b835f661d15462aa0a2b6bde0c9168bb8c18
Author: Gil Forcada <gforcada@gnome.org>
Date:	2012-03-25

    [l10n] Updated Catalan translation

M	po/ca.po

commit 2161aca76cb007e3aaa651546d7c309fac9cb337
Author: Gabor Kelemen <kelemeng@gnome.hu>
Date:	2012-03-25

    Some fixes in the Hungarian help translation

M	help/hu/hu.po

commit cdd07a73147df0ec9b0fe18d69aec0757bda4f9d
Author: Praveen Illa <mail2ipn@gmail.com>
Date:	2012-03-23

    Updated Telugu Transltion

M	po/te.po

commit 287875e8f4a0aea6d9d52417203175a4dccb3736
Author: Kristjan SCHMIDT <kristjan.schmidt@googlemail.com>
Date:	2012-03-22

    Updated Esperanto translation

M	po/eo.po

commit bce0726f1a44b2c131add02da37c54276e278c5b
Author: Kristjan SCHMIDT <kristjan.schmidt@googlemail.com>
Date:	2012-03-22

    Updated Esperanto translation

M	po/eo.po

commit 44004f354728c2ab481c20d16aefe0d10ae27196
Author: Gabor Kelemen <kelemeng@gnome.hu>
Date:	2012-03-21

    Add hu to DOC_LINGUAS

M	help/Makefile.am

commit 27a09499b877ae7cd6e94cf964b99f5db2045a56
Author: Gabor Kelemen <kelemeng@gnome.hu>
Date:	2012-03-20

    Added Hungarian help translation

A	help/hu/hu.po

commit 8b2a285c3be3c2d14821965a148c0ce1694c973a
Author: Gabor Kelemen <kelemeng@gnome.hu>
Date:	2012-03-20

    Updated Hungarian translation

M	po/hu.po

commit 35e659bf4c838652b9a1b06c098cc8a19d8efe8a
Author: Antonio Fernandes C. Neto <fernandesn@gnome.org>
Date:	2012-03-20

    Updated Brazilian Portuguese translation

M	po/pt_BR.po

commit 3a58c74410a5fdab6505cf2506f9dcda04bc1fe4
Author: EL8LatSPQ <el8latspq@hotmail.com>
Date:	2012-03-20

    Update Simplified Chinese translation.

M	po/zh_CN.po

commit 7575e2f36581560566b5c78e5bb9fc364e515017

nemo/nemo-fileroller.h
/*
 *  File-Roller
 * 
 *  Copyright (C) 2004 Free Software Foundation, Inc.
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU General Public
 *  License as published by the Free Software Foundation; either
 *  version 2 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Library General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public
 *  License along with this library; if not, write to the Free
 *  Software Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA 02110-1335, USA.
 *
 *  Author: Paolo Bacchilega <paobac@cvs.gnome.org>
 * 
 */

#ifndef NEMO_FILEROLLER_H
#define NEMO_FILEROLLER_H

#include <glib-object.h>

G_BEGIN_DECLS

#define NEMO_TYPE_FR  (nemo_fr_get_type ())
#define NEMO_FR(o)    (G_TYPE_CHECK_INSTANCE_CAST ((o), NEMO_TYPE_FR, NemoFr))
#define NEMO_IS_FR(o) (G_TYPE_CHECK_INSTANCE_TYPE ((o), NEMO_TYPE_FR))

typedef struct _NemoFr      NemoFr;
typedef struct _NemoFrClass NemoFrClass;

struct _NemoFr {
	GObject __parent;
};

struct _NemoFrClass {
	GObjectClass __parent;
};

GType nemo_fr_get_type      (void);
void  nemo_fr_register_type (GTypeModule *module);

G_END_DECLS

#endif /* NEMO_FILEROLLER_H */


nemo/fileroller-module.c
/*
 *  File-Roller
 * 
 *  Copyright (C) 2004 Free Software Foundation, Inc.
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU General Public
 *  License as published by the Free Software Foundation; either
 *  version 2 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Library General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public
 *  License along with this library; if not, write to the Free
 *  Software Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA 02110-1335, USA.
 *
 *  Author: Paolo Bacchilega <paobac@cvs.gnome.org>
 * 
 */

#include <config.h>
#include <libnemo-extension/nemo-extension-types.h>
#include <libnemo-extension/nemo-column-provider.h>
#include <glib/gi18n-lib.h>
#include "nemo-fileroller.h"


void
nemo_module_initialize (GTypeModule*module)
{
	nemo_fr_register_type (module);

	bindtextdomain (GETTEXT_PACKAGE, GNOMELOCALEDIR);
	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
}


void
nemo_module_shutdown (void)
{
}


void 
nemo_module_list_types (const GType **types,
			    int          *num_types)
{
	static GType type_list[1];
	
	type_list[0] = NEMO_TYPE_FR;
	*types = type_list;
	*num_types = 1;
}


nemo/nemo-fileroller.c
/*
 *  File-Roller
 *
 *  Copyright (C) 2004 Free Software Foundation, Inc.
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU General Public
 *  License as published by the Free Software Foundation; either
 *  version 2 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Library General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public
 *  License along with this library; if not, write to the Free
 *  Software Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA 02110-1335, USA.
 *
 *  Author: Paolo Bacchilega <paobac@cvs.gnome.org>
 *
 */

#include <config.h>
#include <string.h>
#include <glib/gi18n-lib.h>
#include <gio/gio.h>
#include <libnemo-extension/nemo-extension-types.h>
#include <libnemo-extension/nemo-file-info.h>
#include <libnemo-extension/nemo-menu-provider.h>
#include "nemo-fileroller.h"


static GObjectClass *parent_class;


static void
extract_to_callback (NemoMenuItem *item,
		     gpointer          user_data)
{
	GList            *files;
	NemoFileInfo *file;
	char             *uri, *default_dir;
	GString          *cmd;

	files = g_object_get_data (G_OBJECT (item), "files");
	file = files->data;

	uri = nemo_file_info_get_uri (file);
	default_dir = nemo_file_info_get_parent_uri (file);

	cmd = g_string_new ("file-roller");
	g_string_append_printf (cmd,
				" --default-dir=%s --extract %s",
				g_shell_quote (default_dir),
				g_shell_quote (uri));

#ifdef DEBUG
	g_print ("EXEC: %s\n", cmd->str);
#endif

	g_spawn_command_line_async (cmd->str, NULL);

	g_string_free (cmd, TRUE);
	g_free (default_dir);
	g_free (uri);
}


static void
extract_here_callback (NemoMenuItem *item,
		       gpointer          user_data)
{
	GList            *files, *scan;
	NemoFileInfo *file;
	char             *dir;
	GString          *cmd;

	files = g_object_get_data (G_OBJECT (item), "files");
	file = files->data;

	dir = nemo_file_info_get_parent_uri (file);

	cmd = g_string_new ("file-roller");
	g_string_append_printf (cmd," --extract-here");

	g_free (dir);

	for (scan = files; scan; scan = scan->next) {
		NemoFileInfo *file = scan->data;
		char             *uri;

		uri = nemo_file_info_get_uri (file);
		g_string_append_printf (cmd, " %s", g_shell_quote (uri));
		g_free (uri);
	}

	g_spawn_command_line_async (cmd->str, NULL);

#ifdef DEBUG
	g_print ("EXEC: %s\n", cmd->str);
#endif

	g_string_free (cmd, TRUE);
}


static void
add_callback (NemoMenuItem *item,
	      gpointer          user_data)
{
	GList            *files, *scan;
	NemoFileInfo *file;
	char             *uri, *dir;
	GString          *cmd;

	files = g_object_get_data (G_OBJECT (item), "files");
	file = files->data;

	uri = nemo_file_info_get_uri (file);
	dir = g_path_get_dirname (uri);

	cmd = g_string_new ("file-roller");
	g_string_append_printf (cmd," --default-dir=%s --add", g_shell_quote (dir));

	g_free (dir);
	g_free (uri);

	for (scan = files; scan; scan = scan->next) {
		NemoFileInfo *file = scan->data;

		uri = nemo_file_info_get_uri (file);
		g_string_append_printf (cmd, " %s", g_shell_quote (uri));
		g_free (uri);
	}

	g_spawn_command_line_async (cmd->str, NULL);

	g_string_free (cmd, TRUE);
}


static struct {
	char     *mime_type;
	gboolean  is_compressed;
} archive_mime_types[] = {
		{ "application/x-7z-compressed", TRUE },
		{ "application/x-7z-compressed-tar", TRUE },
		{ "application/x-ace", TRUE },
		{ "application/x-alz", TRUE },
		{ "application/x-ar", TRUE },
		{ "application/x-arj", TRUE },
		{ "application/x-bzip", TRUE },
		{ "application/x-bzip-compressed-tar", TRUE },
		{ "application/x-bzip1", TRUE },
		{ "application/x-bzip1-compressed-tar", TRUE },
		{ "application/vnd.ms-cab-compressed", TRUE },
		{ "application/x-cbr", TRUE },
		{ "application/x-cbz", TRUE },
		{ "application/x-cd-image", FALSE },
		{ "application/x-compress", TRUE },
		{ "application/x-compressed-tar", TRUE },
		{ "application/x-cpio", TRUE },
		{ "application/x-deb", TRUE },
		{ "application/x-ear", TRUE },
		{ "application/x-ms-dos-executable", FALSE },
		{ "application/x-gtar", FALSE },
		{ "application/x-gzip", TRUE },
		{ "application/x-gzpostscript", TRUE },
		{ "application/x-java-archive", TRUE },
		{ "application/x-lha", TRUE },
		{ "application/x-lhz", TRUE },
		{ "application/x-lzip", TRUE },
		{ "application/x-lzip-compressed-tar", TRUE },
		{ "application/x-lzma", TRUE },
		{ "application/x-lzma-compressed-tar", TRUE },
		{ "application/x-lzop", TRUE },
		{ "application/x-lzop-compressed-tar", TRUE },
		{ "application/x-ms-wim", TRUE },
		{ "application/x-rar", TRUE },
		{ "application/x-rar-compressed", TRUE },
		{ "application/x-rpm", TRUE },
		{ "application/x-rzip", TRUE },
		{ "application/x-tar", FALSE },
		{ "application/x-tarz", TRUE },
		{ "application/x-stuffit", TRUE },
		{ "application/x-war", TRUE },
		{ "application/x-xz", TRUE },
		{ "application/x-xz-compressed-tar", TRUE },
		{ "application/x-zip", TRUE },
		{ "application/x-zip-compressed", TRUE },
		{ "application/x-zoo", TRUE },
		{ "application/zip", TRUE },
		{ "multipart/x-zip", TRUE },
		{ NULL, FALSE }
};


typedef struct {
      gboolean is_archive;
      gboolean is_derived_archive;
      gboolean is_compressed_archive;
} FileMimeInfo;


static FileMimeInfo
get_file_mime_info (NemoFileInfo *file)
{
	FileMimeInfo file_mime_info;
	int          i;

	file_mime_info.is_archive = FALSE;
	file_mime_info.is_derived_archive = FALSE;
	file_mime_info.is_compressed_archive = FALSE;

	for (i = 0; archive_mime_types[i].mime_type != NULL; i++)
		if (nemo_file_info_is_mime_type (file, archive_mime_types[i].mime_type)) {
			char *mime_type;
			char *content_type_mime_file;
			char *content_type_mime_compare;

			mime_type = nemo_file_info_get_mime_type (file);

			content_type_mime_file = g_content_type_from_mime_type (mime_type);
			content_type_mime_compare = g_content_type_from_mime_type (archive_mime_types[i].mime_type);

			file_mime_info.is_archive = TRUE;
			file_mime_info.is_compressed_archive = archive_mime_types[i].is_compressed;
			if ((content_type_mime_file != NULL) && (content_type_mime_compare != NULL))
				file_mime_info.is_derived_archive = ! g_content_type_equals (content_type_mime_file, content_type_mime_compare);

			g_free (mime_type);
			g_free (content_type_mime_file);
			g_free (content_type_mime_compare);

			return file_mime_info;
		}

	return file_mime_info;
}


static gboolean
unsupported_scheme (NemoFileInfo *file)
{
	gboolean  result = FALSE;
	GFile    *location;
	char     *scheme;

	location = nemo_file_info_get_location (file);
	scheme = g_file_get_uri_scheme (location);

	if (scheme != NULL) {
		const char *unsupported[] = { "trash", "computer", NULL };
		int         i;

		for (i = 0; unsupported[i] != NULL; i++)
			if (strcmp (scheme, unsupported[i]) == 0)
				result = TRUE;
	}

	g_free (scheme);
	g_object_unref (location);

	return result;
}


static GList *
nemo_fr_get_file_items (NemoMenuProvider *provider,
			    GtkWidget            *window,
			    GList                *files)
{
	GList    *items = NULL;
	GList    *scan;
	gboolean  can_write = TRUE;
	gboolean  one_item;
	gboolean  one_archive = FALSE;
	gboolean  one_derived_archive = FALSE;
	gboolean  one_compressed_archive = FALSE;
	gboolean  all_archives = TRUE;
	gboolean  all_archives_derived = TRUE;
	gboolean  all_archives_compressed = TRUE;

	if (files == NULL)
		return NULL;

	if (unsupported_scheme ((NemoFileInfo *) files->data))
			return NULL;

	for (scan = files; scan; scan = scan->next) {
		NemoFileInfo *file = scan->data;
		FileMimeInfo      file_mime_info;

		file_mime_info = get_file_mime_info (file);

		if (all_archives && ! file_mime_info.is_archive)
			all_archives = FALSE;

		if (all_archives_compressed && file_mime_info.is_archive && ! file_mime_info.is_compressed_archive)
			all_archives_compressed = FALSE;

		if (all_archives_derived && file_mime_info.is_archive && ! file_mime_info.is_derived_archive)
			all_archives_derived = FALSE;

		if (can_write) {
			NemoFileInfo *parent;

			parent = nemo_file_info_get_parent_info (file);
 			can_write = nemo_file_info_can_write (parent);
		}
	}

	/**/

	one_item = (files != NULL) && (files->next == NULL);
	one_archive = one_item && all_archives;
	one_derived_archive = one_archive && all_archives_derived;
	one_compressed_archive = one_archive && all_archives_compressed;

	if (all_archives && can_write) {
		NemoMenuItem *item;

		item = nemo_menu_item_new ("NemoFr::extract_here",
					       _("Extract Here"),
					       /* Translators: the current position is the current folder */
					       _("Extract the selected archive to the current position"),
					       "drive-harddisk");
		g_signal_connect (item,
				  "activate",
				  G_CALLBACK (extract_here_callback),
				  provider);
		g_object_set_data_full (G_OBJECT (item),
					"files",
					nemo_file_info_list_copy (files),
					(GDestroyNotify) nemo_file_info_list_free);

		items = g_list_append (items, item);
	}
	else if (all_archives && ! can_write) {
		NemoMenuItem *item;

		item = nemo_menu_item_new ("NemoFr::extract_to",
					       _("Extract To..."),
					       _("Extract the selected archive"),
					       "drive-harddisk");
		g_signal_connect (item,
				  "activate",
				  G_CALLBACK (extract_to_callback),
				  provider);
		g_object_set_data_full (G_OBJECT (item),
					"files",
					nemo_file_info_list_copy (files),
					(GDestroyNotify) nemo_file_info_list_free);

		items = g_list_append (items, item);

	}

	if (! one_compressed_archive || one_derived_archive) {
		NemoMenuItem *item;

		item = nemo_menu_item_new ("NemoFr::add",
					       _("Compress..."),
					       _("Create a compressed archive with the selected objects"),
					       "gnome-mime-application-x-archive");
		g_signal_connect (item,
				  "activate",
				  G_CALLBACK (add_callback),
				  provider);
		g_object_set_data_full (G_OBJECT (item),
					"files",
					nemo_file_info_list_copy (files),
					(GDestroyNotify) nemo_file_info_list_free);

		items = g_list_append (items, item);
	}

	return items;
}


static void
nemo_fr_menu_provider_iface_init (NemoMenuProviderIface *iface)
{
	iface->get_file_items = nemo_fr_get_file_items;
}


static void
nemo_fr_instance_init (NemoFr *fr)
{
}


static void
nemo_fr_class_init (NemoFrClass *class)
{
	parent_class = g_type_class_peek_parent (class);
}


static GType fr_type = 0;


GType
nemo_fr_get_type (void)
{
	return fr_type;
}


void
nemo_fr_register_type (GTypeModule *module)
{
	static const GTypeInfo info = {
		sizeof (NemoFrClass),
		(GBaseInitFunc) NULL,
		(GBaseFinalizeFunc) NULL,
		(GClassInitFunc) nemo_fr_class_init,
		NULL,
		NULL,
		sizeof (NemoFr),
		0,
		(GInstanceInitFunc) nemo_fr_instance_init,
	};

	static const GInterfaceInfo menu_provider_iface_info = {
		(GInterfaceInitFunc) nemo_fr_menu_provider_iface_init,
		NULL,
		NULL
	};

	fr_type = g_type_module_register_type (module,
					       G_TYPE_OBJECT,
					       "NemoFileRoller",
					       &info, 0);

	g_type_module_add_interface (module,
				     fr_type,
				     NEMO_TYPE_MENU_PROVIDER,
				     &menu_provider_iface_info);
}
